.tekton/single-arch-build-pipeline.yaml

```diff
@@ -0,0 +1,586 @@
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: single-arch-build-pipeline
+spec:
+  tasks:
```

Comment:
Can we add work spaces to the flowing tasks ? 
Upstream refrence: https://github.com/konflux-ci/build-definitions/blob/main/pipelines/docker-build-oci-ta/docker-build-oci-ta.yaml
  - sast-snyk-check (upstream line 316)
  - sast-coverity-check (upstream line 376)
  - sast-shell-check (upstream line 408)
  - sast-unicode-check (upstream line 429)
  - push-dockerfile (upstream line 458)